### PR TITLE
Make ThreadIdInner a struct.

### PR DESCRIPTION
**Pull request by ltratt**
This makes it slightly harder to accidentally cast the wrong thing.

### CODE REVIEW

File: ykrt/src/mt.rs [hunk 1]
```diff
@@ -235,6 +236,7 @@ impl MTInner {
 }
 
 type CompilingTrace<I> = Mutex<Option<Box<CompiledTrace<I>>>>;
+struct ThreadIdInner(u8);
```
Comment (bjorn3) [inline ykrt/src/mt.rs:239]: I think this can be a unit struct. The `Arc` will have to allocate memory any way to fit the strong and weak reference count.

Comment (ltratt) [reply ykrt/src/mt.rs:239]: I remember reading that there are some `malloc`s somewhere that don't actually allocate memory if you ask for 0 bytes, so I'm a bit nervous about that. i also half-wondered if I should use a `usize` here to force word-level alignment, but that might be unduly paranoid of me!

Comment (bjorn3) [reply ykrt/src/mt.rs:239]: The `Arc` itself will allocate a memory locatikn big enough to fit the pointer sized strong and weak reference counters before the actual data. https://github.com/rust-lang/rust/blob/2ba7ca2bbbff6cd424aebc654308febc00b9497a/library/alloc/src/sync.rs#L301

Comment (ltratt) [reply ykrt/src/mt.rs]: Good point! Fixed in https://github.com/softdevteam/yk/pull/195/commits/3d21b29494100c9b7a62c71c1cc59c3fe5685681.

**Review by bjorn3 — APPROVED** [review]

**Comment by vext01** [comment]
Can you expand on the PR description? I don't understand how the change makes it harder to accidentally cast the wrong thing.

**Comment by ltratt** [comment]
It's just a standard "new type" thing: I can't confuse a new type with a common type (in other words, if I have another `Arc<u8>` somewhere serving a different purpose, I won't confuse that with a `Arc<ThreadIdInner>`).

**Comment by vext01** [comment]
Ah, gotcha! Yes, that's a mighty fine idea.

Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**